webapi/src/main/scala/org/knora/webapi/messages/admin/responder/permissionsmessages/PermissionsMessagesADM.scala

```diff
@@ -697,6 +703,30 @@ case class PermissionByIriGetRequestADM(permissionIri: IRI, requestingUser: User
   stringFormatter.validatePermissionIri(permissionIri,
                                         throw BadRequestException(s"Invalid permission IRI $permissionIri is given."))
 }
+
+/**
+  * A message that requests deletion of a permission identified through its IRI.
+  * A successful response will be [[PermissionDeleteResponseADM]] with deleted=true.
+  *
+  * @param permissionIri               the iri of the permission object.
+  * @param requestingUser              the user initiating the request.
+  * @param apiRequestID                the API request ID.
+  */
+case class PermissionDeleteRequestADM(permissionIri: IRI, requestingUser: UserADM, apiRequestID: UUID)
+    extends PermissionsResponderRequestADM {
+  // Check user's permission for the operation
+  if (!requestingUser.isSystemAdmin
+      && !requestingUser.permissions.isProjectAdminInAnyProject()
```

Comment:
Isn't it necessary for the `requestingUser` to be `ProjectAdmin` of the project that the permission belongs to?

Reply:
Thanks for pointing this out, I limited this to project admin.